src/main/java/org/folio/service/consortium/SharingInstanceService.java

```diff
@@ -67,4 +71,23 @@ private Future<SharingInstance> shareInstance(String consortiumId, SharingInstan
       });
   }
 
+  /**
+   * Gets sharing instances. The result will include information in which tenants instances already existed.
+   *
+   * @param instanceId  the instance id
+   * @param consortiumConfiguration the consortium configuration
+   * @param requestContext the request context
+   * @return list with shared instances
+   */
+  public Future<SharingInstanceCollection> getSharingInstances(String instanceId,
+                                                               ConsortiumConfiguration consortiumConfiguration,
+                                                               RequestContext requestContext) {
+    RequestEntry requestEntry = new RequestEntry(SHARE_INSTANCE_ENDPOINT)
```

Comment:
**Question**
How is this implemented? Does it iterates over all tenants and tries to obtain instance with given id?

Reply:
We have table in mod-consortia central tenant sharing_instance that knows all about tenants where shared instance exist, so we use it here.
Here is detailed design diagram: https://folio-org.atlassian.net/wiki/spaces/DD/pages/1781565/Sharing+Instances+With+Consortium+Members